sdk/samples/iot/docs/how_to_iot_hub_esp32.md

```diff
@@ -0,0 +1,311 @@
+        Writing at 0x00038000... (83 %)
+        Writing at 0x0003c000... (88 %)
+        Writing at 0x00040000... (94 %)
+        Writing at 0x00044000... (100 %)
+        Wrote 396432 bytes (292339 compressed) at 0x00000000 in 13.0 seconds (effective 243.4 kbit/s)...
+        Hash of data verified.
+
+        Leaving...
+        Hard resetting via RTS pin...
+        ```
+
+        </p>
+        </details>
+
+8. Monitor the MCU (microcontroller) locally via the Serial Port.
+
+    - Go to menu `Tools`, `Serial Monitor`.
+
+        If you perform this step right away after uploading the sketch, the serial monitor will show an output similar to the following upon success:
+
+        ```text
+        Connecting to WIFI SSID buckaroo
+        .......................WiFi connected, IP address:
+        192.168.1.123
+        Setting time using SNTP..............................done!
+        Current time: Thu May 28 02:55:05 2020
+        Client ID: mydeviceid
+        Username: myiothub.azure-devices.net/mydeviceid/?api-version=2018-06-30&DeviceClientType=c%2F1.0.0
+        Password: SharedAccessSignature sr=myiothub.azure-devices.net%2Fdevices%2Fmydeviceid&sig=0VFwGiXlIVPeCmPStJ4Fb1wbS8o2W8p1vzIOt%2B8K2eE%3D&se=1590620105
+        MQTT connecting ... connected.
+        ```
+
+9. Monitor the telemetry messages sent to the Azure IoT Hub.
+
+    ```bash
+    $ az iot hub monitor-events --login <your Azure IoT Hub connection string in quotes> --device-id <your device id>
+    ```
+
+    <details><summary><i>Expected telemetry output:</i></summary>
+    <p>
+
+    ```bash
+    Starting event monitor, filtering on device: mydeviceid, use ctrl-c to stop...
+    {
+        "event": {
+            "origin": "mydeviceid",
+            "payload": "payload"
+        }
+    }
+    {
+        "event": {
+            "origin": "mydeviceid",
+            "payload": "payload"
+        }
+    }
+    {
+        "event": {
+            "origin": "mydeviceid",
+            "payload": "payload"
+        }
+    }
+    {
+        "event": {
+            "origin": "mydeviceid",
+            "payload": "payload"
+        }
+    }
+    {
+        "event": {
+            "origin": "mydeviceid",
+            "payload": "payload"
+        }
+    }
+    {
+        "event": {
+            "origin": "mydeviceid",
+            "payload": "payload"
+        }
+    }
+    ^CStopping event monitor...
+    ```
+
+    </p>
+    </details>
+
+## Certificates - Important to know
+
+The Azure IoT service certificates presented during TLS negotiation shall be always validated, on the device, using the appropriate trusted root CA certificate(s).
+
+For the ESP32 sample, our script `generate_arduino_zip_library.sh` automatically downloads the root certificate used in the United States regions (Baltimore CA certificate) and adds it to the Arduino sketch project.
+
+For other regions (and private cloud environments), please use the appropriate root CA certificate.
```

Comment:
should we link to the certs we have in `certs.c` of V1?

Reply:
No, we are not doing that way anymore. We should not have certs checked into the repos going forward.